### PR TITLE
migrating popup item media types to image, video, audio, 3d_model & pdf

### PR DESCRIPTION
**Pull request by ratankhalsa92**

### CODE REVIEW

File: src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * Migrating media type for popup items
+ * Old Types: media, youtube, embed_code, sprite
+ * New Types: image, video, audio, 3d_model, pdf, youtube, markdown, embed_code, sprite
+ */
+class Migration implements IOrder {
+  execute(projectJson: unknown) {
+    const pJson = projectJson as RecordNode<RT.project>;
+    const projectF = r.record(pJson);
+
+    const allPopupElements = projectF.getAllDeepChildrenWithFilter(RT.element, el => el.props.element_type === "popup");
```
Comment (amitrajput1992) [inline src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts:15]: `el.props.element_type === en.ElementType.popup`

File: src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        let sourceUri = "";
+
+        //* It's just the "media" type that will be split into image, video, audio, pdf or 3d_model
+        if(item.props.item_source_type === "media") {
+          if(typeof source === "object" && source.file_urls) {
+            sourceUri = source.file_urls ? (source.file_urls.gltf ?? source.file_urls.o): "";
+          }
+        }
+
+        const { getFileType, FileType } = pathUtils;
```
Comment (amitrajput1992) [inline src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts:32]: First check should be on `source.type` which will be one of the enum values from `FileType`
secondary check should be done as a failsafe, this being the file extension check.

**Review by amitrajput1992 — CHANGES_REQUESTED** [review]

File: src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts (hunk 2 above)
Comment (ratankhalsa92) [reply src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts:32]: Checked on line 26.

File: src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    const projectF = r.record(pJson);
+
+    const allPopupElements = projectF.getAllDeepChildrenWithFilter(RT.element, el => el.props.element_type === "popup");
+
+    for (const element of allPopupElements) {
+      const elementF = r.element(element);
+      const allItems = elementF.getRecords(RT.item);
+
+      for(const item of allItems) {
+        const source = item.props.item_source as fn.Source;
```
Comment (amitrajput1992) [inline src/migrations/project/r-migration-commands/m148_149_migrate_popup_item_media_types.ts:22]: const itemF = r.record(item);
const source = itemF.getValueOrDefault(rtp.item.item_source) as fn.Source;
let sourceType = source.type as pathUtils.FileType;
if(!sourceType) {
  // get the file type from the file_urls
  const url = source.file_urls?.glb || source.file_urls?.gltf || source.file_urls?.o || "";
  sourceType = pathUtils.getFileType(url);
}

// switch-case

**Review by amitrajput1992 — APPROVED** [review]

**Pull request merged**